File: backend/internal/models/customer.go
package models

import (
	"time"
)

// Customer represents a client company
type Customer struct {
	CustomerID  int       `db:"customer_id" json:"customer_id"`
	CompanyName string    `db:"company_name" json:"company_name"`
	Industry    *string   `db:"industry" json:"industry,omitempty"`
	Address     *string   `db:"address" json:"address,omitempty"`
	Phone       *string   `db:"phone" json:"phone,omitempty"`
	Email       *string   `db:"email" json:"email,omitempty"`
	Website     *string   `db:"website" json:"website,omitempty"`
	CreatedAt   time.Time `db:"created_at" json:"created_at"`
	UpdatedAt   time.Time `db:"updated_at" json:"updated_at"`
}

File: frontend/src/types/Customer.ts
export interface Customer {
  customer_id: number;
  company_name: string;
  industry?: string;
  address?: string;
  phone?: string;
  email?: string;
  website?: string;
  created_at: string;
  updated_at: string;
}

export interface CustomerCreate extends Omit<Customer, 'customer_id' | 'created_at' | 'updated_at'> {
  customer_id?: number;
}

export interface CustomerUpdate extends CustomerCreate {
  customer_id: number;
} 